### DISCUSSION TITLE
Virtual Machine for node.js application

### DISCUSSION THREAD

**Discussion by tomasludv** [community/community#61663] [Programming Help]
Hello, what is the best and cheapest way to run node.js application that runs 24/7. It's not web application.

Virtual Machine on Azure, AWS, Google?
Docker on Azure, AWS, Google?
Raspberry pi?
Is there some cloud service that can run a simple node.js app 24/7 without using VM, Docker, Raspberry?

**Comment by Beargruug** [comment]
Hey there!

1. Virtual Machine on Azure, AWS, Google Cloud:
One way to go is setting up your Node.js app on a virtual machine provided by Azure, AWS, or Google Cloud. It's like having your own computer in the cloud. This gives you full control over the environment, but keep in mind it might not be the cheapest option, especially if your app isn't very resource-demanding.

2. Docker on Azure, AWS, Google Cloud:
If you're into containerization, Docker could be the way to go. With Docker, you can package your Node.js app and all its dependencies into a single container. This makes deployment and scaling easier, and it might be more cost-effective than using a full virtual machine.

3. Raspberry Pi:
For smaller projects or if you prefer a local solution, you can run your Node.js app on a Raspberry Pi. It's a low-cost option, but it might have limitations in terms of computing power and reliability compared to cloud-based solutions.

4. Serverless Platforms:
Some cloud providers offer serverless platforms like AWS Lambda or Azure Functions. With serverless, you don't have to manage a server; you just pay for the actual compute time used. It can be cost-efficient for low to moderate traffic apps, but keep in mind there are execution time limits and potential cold start delays.

5. PaaS (Platform-as-a-Service) Offerings:
Consider platforms like Azure App Service or Google App Engine. They abstract away infrastructure management, making it easy to deploy your Node.js app without worrying about servers. However, they might be less customizable and potentially more expensive than managing VMs or containers directly.

To find the best and cheapest option for your app, think about its resource requirements, expected traffic, scalability needs, and budget.
